src/runtime/manifest-primitive-parser/manifest-primitive-parser.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * @license
+ * Copyright (c) 2017 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Loader} from '../../platform/loader.js';
+import {parse as parser} from '../../gen/runtime/manifest-parser.js';
+
+import * as AstNode from '../manifest-ast-types/manifest-ast-nodes.js';
+import {Dictionary} from '../../utils/lib-utils.js';
+
+export type Ast = AstNode.All[];
+export {AstNode};
+
+interface ManifestLoadOptions {
+  registry?: Dictionary<Promise<Ast>>;
+  errors?: ManifestError[];
+}
+
+export interface ManifestParseOptions extends ManifestLoadOptions {
+  filename?: string;
+  loader?: Loader;
```

Comment:
All these `loader`s are notionally the same. If I make the field `importLoader` then I will have to assign `loader` to it. I can change it, but I don't really understand the objection.

Reply:
I was a bit confused when first reading through; why did we need a loader arg and a loader option? Once I saw how they were used it was understandable, but users of an API shouldn't need to see the code to avoid confusion.

It's not a big deal, but even a comment on the loader field would help. (It's obvious why a call to `load` would need a loader.)

Reply:
Added